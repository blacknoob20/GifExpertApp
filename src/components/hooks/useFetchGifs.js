import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (categoria) => {
    const [state, setState] = useState({
        data:[],
        losding: true
    });

    useEffect(()=>{ console.log(categoria);
        getGifs(categoria)
          .then(imgs => setState({
              data: imgs,
              loading: false
          }));
    },[categoria]);

    return state;
}
