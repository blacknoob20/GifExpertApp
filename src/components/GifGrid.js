// import React, { Fragment, useEffect, useState } from 'react';
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
// import { getGifs } from './helpers/getGifs';
import { useFetchGifs } from './hooks/useFetchGifs';

export const GifGrid = ({categoria}) => {
    // const [imgs, setImgs] = useState([]);
    const {data:imgs,loading} = useFetchGifs(categoria);

    // useEffect(()=>{
    //     getGifs(categoria)
    //         .then(setImgs);
    // }, [categoria]);

    // getGifs();

        return(
            <Fragment>
                <h3>{categoria}</h3>
                {loading && <p className="animate_animated animate_fadeIn">Loading</p>}
                <div className="card-grid">
                    { imgs.map((img) =>(<GifGridItem key={img.id} {...img}/>))}
                </div>
            </Fragment >
        );
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired,
}