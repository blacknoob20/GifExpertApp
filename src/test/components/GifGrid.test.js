import React from 'react';
import { shallow } from 'enzyme/build';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../components/hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../components/hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {
    const categoria = 'One Punch';

    test('Debe mostrarse correctamente <GifGrid />', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid categoria={categoria} />);

        expect(wrapper).toMatchSnapshot();
    });


    test('Debe mostrarse items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const imgs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa',
        }]

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: true,
        });

        const wrapper = shallow(<GifGrid categoria={categoria} />);

        expect( wrapper.find('p').exist() ).toBe(false);
        // expect( wrapper.find('GifGridItem').length ).toBe(imgs.length);
    });
});
