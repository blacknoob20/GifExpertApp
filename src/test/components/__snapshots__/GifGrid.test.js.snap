// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe mostrarse correctamente <GifGrid /> 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate_animated animate_fadeIn"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
